java/vector/src/main/java/org/apache/arrow/vector/NullableTimestampVector.java

```diff
@@ -124,6 +177,32 @@ public void setSafe(int index, long value) {
     set(index, value);
   }
 
+  public void setSafe(int index, NullableTimestampHolder holder) {
```

Comment:
Do these set methods need to check that the type params are compatible?

Reply:
Good point. Although it does add branching on the array. Maybe don't check for performance reason, not sure.